cmake/reports/CMakeLists.txt

```diff
@@ -34,3 +34,22 @@ if(NOT ${PUNCOVER} STREQUAL PUNCOVER-NOTFOUND)
     USES_TERMINAL
     )
 endif()
+
+find_program(PAHOLE pahole)
+
+if(NOT ${PAHOLE} STREQUAL PAHOLE-NOTFOUND)
+  add_custom_target(
+    pahole
+    ${PAHOLE}
+    --anon_include
+    --nested_anon_include
+    --show_decl_info
+    $<$<BOOL:${CMAKE_VERBOSE_MAKEFILE}>:--verbose>
+    ${ZEPHYR_BINARY_DIR}/${KERNEL_ELF_NAME}
+    > ${ZEPHYR_BINARY_DIR}/${KERNEL_NAME}.pahole
```

Comment:
can this go on the screen instead and have the caller redirect to a file if they want to?

Reply:
Certainly, if that is preferred. It's usually a lot of output, though.

Reply:
Updated.